src/plugins/gui/providers/ioniaProvider.ts

```diff
@@ -277,30 +277,28 @@ export const makeIoniaProvider: FiatProviderFactory<IoniaMethods> = {
     //
 
     let hiddenCardIds: number[] = asStoreHiddenCards(await store.getItem(STORE_HIDDEN_CARDS_KEY).catch(_ => undefined))
-    let purchaseCardPromise: Promise<IoniaPurchaseCard> // There can only be one!
+    let purchaseCardTimeoutId: NodeJS.Timeout
     const ratesCache: { [currencyCode: string]: { expiry: number; rateQueryPromise: Promise<number> } } = {}
 
     //
     // Private methods:
     //
 
     async function getPurchaseCard(currencyCode: string, cardAmount: number): Promise<IoniaPurchaseCard> {
-      const newPurchaseCardPromise = queryPurchaseCard(currencyCode, cardAmount)
-      purchaseCardPromise = newPurchaseCardPromise
-
-      // Hastily resolve the promise with a debounce:
-      setTimeout(
-        () => {
-          if (purchaseCardPromise === newPurchaseCardPromise) {
-            newPurchaseCardPromise.then(() => {
-              // IoniaPurchaseCard ready!
-            })
+      return await new Promise<IoniaPurchaseCard>((resolve, reject) => {
+        // Clear the previous task timeout if present
+        if (purchaseCardTimeoutId != null) clearTimeout(purchaseCardTimeoutId)
+
+        // Hastily invoke the task promise with a debounce:
+        const newPurchaseCardTimeoutId = setTimeout(() => {
+          if (purchaseCardTimeoutId === newPurchaseCardTimeoutId) {
+            queryPurchaseCard(currencyCode, cardAmount).then(resolve, reject)
```

Comment:
the `else` case leaves a dangling promise. I think we're guilty of this elsewhere in the GUI but ideally return undefined and have caller of `getPurchaseCard` throw it away.